tests/ui/diagnostic-flags/allow-non-lint-warnings.rs

```diff
@@ -0,0 +1,20 @@
+// ignore-tidy-linelength
+//! Check that `-A warnings` cli flag applies to *all* warnings, including feature gate warnings.
+//!
+//! This test tries to exercise that by checking that the "empty trait list for derive" warning for
+//! `#[derive()]` is permitted by `-A warnings`, which is a non-lint warning.
```

Comment:
Remark: I elaborated on the "how" here because it was very not obvious how this test did the check.